app/assets/javascripts/discourse/tests/unit/lib/search-test.js

```diff
@@ -62,4 +63,190 @@ module("Unit | Utility | search", function (hooks) {
+        id: 246,
+        name: "The Joker",
+        username: "joker",
+        topic_id: 93,
+        topic: {
+          id: 93,
+          title: "Why don't you put a smile on that face...",
+        },
+      },
+      {
+        id: 104,
+        name: "Steve Rogers",
+        username: "captain_america",
+        topic_id: 2,
+        topic: {
+          id: 2,
+          title: "What its like being frozen...",
+        },
+      },
+      {
+        id: 245,
+        name: "Loki",
+        username: "loki",
+        topic_id: 92,
+        topic: {
+          id: 92,
+          title: "There is only one person you can trust",
+        },
+      },
+    ];
+
+    const desiredMixedResults = [
+      {
+        id: 104,
+        itemKey: "2_2",
+        name: "Steve Rogers",
+        reciprocalRank: 0.30952380952380953,
+        topic: {
+          id: 2,
+          title: "What its like being frozen...",
+        },
+        topic_id: 2,
+        username: "captain_america",
+      },
+      {
+        id: 250,
+        itemKey: "96_96",
+        name: "Bruce Wayne",
+        reciprocalRank: 0.2,
+        topic: {
+          id: 96,
+          title: "I like to fight crime",
+        },
+        topic_id: 96,
+        username: "batman",
+      },
+      {
+        id: 104,
+        itemKey: "95_95",
+        name: "Tony Stark",
+        reciprocalRank: 0.2,
+        topic: {
+          id: 95,
+          title: "What I learned from my father",
+        },
+        topic_id: 95,
+        username: "ironman",
+      },
+      {
+        id: 246,
+        itemKey: "93_93",
+        name: "The Joker",
+        reciprocalRank: 0.16666666666666666,
+        topic: {
+          id: 93,
+          title: "Why don't you put a smile on that face...",
+        },
+        topic_id: 93,
+        username: "joker",
+      },
+      {
+        id: 202,
+        itemKey: "32_32",
+        name: "Peter Parker",
+        reciprocalRank: 0.14285714285714285,
+        topic: {
+          id: 32,
+          title: "My experience meeting the Avengers",
+        },
+        topic_id: 32,
+        username: "spidey",
+      },
+      {
+        id: 290,
+        itemKey: "111_111",
+        name: "Clark Kent",
+        reciprocalRank: 0.125,
+        topic: {
+          id: 111,
+          title: "My fear of Kryptonite",
+        },
+        topic_id: 111,
+        username: "superman",
+      },
+      {
+        id: 245,
+        itemKey: "92_92",
+        name: "Loki",
+        reciprocalRank: 0.125,
+        topic: {
+          id: 92,
+          title: "There is only one person you can trust",
+        },
+        topic_id: 92,
+        username: "loki",
+      },
+    ];
+
+    const rankedList = reciprocallyRankedList(
+      [sourceA, sourceB],
+      ["topic_id", "topic_id"]
+    );
+
+    assert.deepEqual(
+      rankedList,
+      desiredMixedResults,
+      "it correctly ranks the results using the reciprocal ranking algorithm"
+    );
+  });
```

Comment:
Can you add a test case with 3 or 4 lists? I also wonder how the ranking algorithm handles different data stuctures in lists, maybe add a bit of variety in there as well.